src/main/java/core/basesyntax/strategy/ReturnOperationHandler.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.serviceimpl.FruitTransactionValidatorImpl;
+import java.util.Optional;
+
+public class ReturnOperationHandler implements OperationHandler {
+    private FruitTransactionValidatorImpl validator;
+
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
+        validator.validateOperation(fruitTransaction);
+        String fruit = fruitTransaction.getFruit();
+        validator.validateFruit(fruit);
```

Comment:
The call to `validateFruit` seems unnecessary here as the fruit should already be validated at the point of reading the input or earlier in the process.